### PR TITLE
Modify app labels to use app.kubernetes.instance

### LINKED ISSUE #179: Modify some of the app labels
**As-is**:  we set `app.kubernetes.io/name` to be the same as `metadata.name` and don't allow that to be overridden. 

**To-be**: set `app.kubernetes.io/instance` to be the same as `metadata.name` and let `app.kubernetes.io/name` be configurable by the CR (default could still be `metadata.name`)

### PR DESCRIPTION
**Pull request by edavidj**
**What this PR does / why we need it?**:
- Updates our app labels so that we're setting `app.kubernetes.io/instance` to `metadata.name` rather than `app.kubernetes.io/name` which is now user configurable.
- Updates label selectors to use the corresponding instance label rather than name.

**Does this PR introduce a user-facing change?**

- [x] User guide
- [x] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Fixes #179 

**Notes**:
- For a lot of the label selector changes I was working off the assumption that `app.kubernetes.instance` is replacing the functionality of `app.kubernetes.name` as far as operator functionality goes. If this isn't desired let me know.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
+## [0.3.0]
+
+### Changed
+
+- Changed the label corresponding to `metadata.name` to `app.kubernetes.instance` and made former label `app.kubernetes.name` user configurable. ([#179](https://github.com/appsody/appsody-operator/issues/179))
```
Comment (leochr) [inline CHANGELOG.md]: Please specify the full labels here (`app.kubernetes.io/instance` and `app.kubernetes.io/name`)

File: pkg/apis/appsody/v1beta1/appsodyapplication_types.go [hunk 2]
```diff
@@ -516,8 +516,9 @@ func (cr *AppsodyApplication) applyConstants(defaults AppsodyApplicationSpec, co
 
 // GetLabels returns set of labels to be added to all resources
 func (cr *AppsodyApplication) GetLabels() map[string]string {
+
 	labels := map[string]string{
-		"app.kubernetes.io/name":       cr.Name,
+		"app.kubernetes.io/instance":   cr.Name,
```
Comment (leochr) [inline pkg/apis/appsody/v1beta1/appsodyapplication_types.go]: Suggestion: To simplify the code, `app.kubernetes.io/name` can be set to `cr.Name` here by default (just as before). Users can override it later when we go through the list of CR labels.  This way we don't need to check at the end if `app.kubernetes.io/name` is set or not.

File: CHANGELOG.md (hunk 1 above)
Comment (edavidj) [reply CHANGELOG.md]: Ah good catch, will fix 👍

File: pkg/apis/appsody/v1beta1/appsodyapplication_types.go (hunk 2 above)
Comment (edavidj) [reply pkg/apis/appsody/v1beta1/appsodyapplication_types.go]: Oh yeah that would be a lot more concise, thanks I'll add that change now.

**Pull request merged**